RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitMultiFitAlgo.h

```diff
@@ -25,17 +25,27 @@ class EcalUncalibRecHitMultiFitAlgo
   
   EcalUncalibRecHitMultiFitAlgo();
   ~EcalUncalibRecHitMultiFitAlgo() { };
-  EcalUncalibratedRecHit makeRecHit(const EcalDataFrame& dataFrame, const EcalPedestals::Item * aped, const EcalMGPAGainRatio * aGain, const SampleMatrix &noisecor, const FullSampleVector &fullpulse, const FullSampleMatrix &fullpulsecov, const BXVector &activeBX);
+  EcalUncalibratedRecHit makeRecHit(const EcalDataFrame& dataFrame, const EcalPedestals::Item * aped, const EcalMGPAGainRatio * aGain, const SampleMatrixGainArray &noisecors, const FullSampleVector &fullpulse, const FullSampleMatrix &fullpulsecov, const BXVector &activeBX);
   void disableErrorCalculation() { _computeErrors = false; }
   void setDoPrefit(bool b) { _doPrefit = b; }
   void setPrefitMaxChiSq(double x) { _prefitMaxChiSq = x; }
+  void setDynamicPedestals(bool b=true) { _dynamicPedestals = b; }
+  void setMitigateBadSamples(bool b=true) { _mitigateBadSamples = b; }
+  void setSelectiveBadSampleCriteria(bool b=true) { _selectiveBadSampleCriteria = b; }
```

Comment:
please remove default values or introduce set and unset methods if no-argument calls are desired.